docs/src/modules/spring/pages/access-control.adoc

```diff
@@ -1 +1,201 @@
+
+
+== Backoffice and self invocations
+
+Invocations of methods from the same service, or from the backoffice using the `kalix service proxy` command, are always permitted, regardless of what ACLs are defined on them.
+
+== Local development with ACLs
+
+When testing or running in development, by default, all calls to your service will be considered to have come from the internet. You can impersonate a local service by setting the `Impersonate-Kalix-Service` header on the requests you make.
+
+=== Disabling ACLs in local development
+
+When running a service during local development, it may be convenient to turn ACL checking off. This can be done by adding the `ACL_ENABLED` environment variable and setting it to `false` in your `docker-compose.yml` file:
+
+```yaml
+kalix-proxy:
+  image: gcr.io/kalix-public/kalix-proxy:latest
+  command: -Dconfig.resource=dev-mode.conf -Dkalix.proxy.eventing.support=google-pubsub-emulator
+  ports:
+    - "9000:9000"
+  environment:
+    USER_FUNCTION_HOST: ${USER_FUNCTION_HOST:-host.docker.internal}
+    USER_FUNCTION_PORT: 8080
+    ACL_ENABLED: false
+```
+
+=== Service identification in local development
+
+If running multiple services in local development, you may want to run with ACLs enabled to verify that they work for cross-service communication. In order to do this, you need to ensure that when services communicate with each other, they are able to identify themselves to one another. This can be done by setting the `SERVICE_NAME` environment variable in your `docker-compose.yml` file:
+
+```yaml
+kalix-proxy:
+  image: gcr.io/kalix-public/kalix-proxy:latest
+  command: -Dconfig.resource=dev-mode.conf -Dkalix.proxy.eventing.support=google-pubsub-emulator
+  ports:
+    - "9000:9000"
+  environment:
+    USER_FUNCTION_HOST: ${USER_FUNCTION_HOST:-host.docker.internal}
+    USER_FUNCTION_PORT: 8080
+    SERVICE_NAME: my-service-name
+```
+
+Note that in local development, the services don't actually authenticate with each other, they only pass their identity in a header. It is assumed in local development that a client can be trusted to set that header correctly.
+
+== Programmatically accessing principals
+
+The current principal associated with a request can be accessed by reading metadata headers. If the request came from another service, the `_kalix-src-svc` header will be set to the name of the service that made the request. Kalix guarantees that this header will only be present from an authenticated principal, it can't be spoofed.
+
+For internet, self and backoffice requests, the `_kalix-src` header will be set to `internet`, `self` and `backoffice` respectively. Backoffice requests are requests that have been made using the `kalix service proxy` command, they are authenticated and authorized to ensure only developers of your project can make them.
```

Comment:
I think all of this could be moved to a partial and shared with the java/scala sdk right? Seems exactly the same and can't see it changing any time...

Reply:
As a principle is a good idea the problem is that this content reappears in the `kalix-docs`, not in the `kalix-jvm-sdk/docs/java`. The Spring ACL doc is a mix of those two. Making a partial here wouldn't have any benefit. There is no other place to share it.

Reply:
Oh I see.. But in that case, this is an example of a kalix-docs section that will need to be moved down.. I mean it doesn't make sense anymore to have the top-level ["Using ACLs"](https://docs.kalix.io/services/using-acls.html) which mentions gRPC... it needs to become one for Java/Scala and Javascript and another for Spring I guess. And in that case, this "Local development with ACLs" can still be shared. We'll probably need to sync with @beritou on that maybe

Reply:
Fair enough 👍